### PR TITLE
update the lint set with an initial list of lints

### PR DESCRIPTION
**Pull request by devoncarew**
- update the lint set with an initial list of lints

Taken from a doc analyzing the current lints used in the dart.dev packages.

### CODE REVIEW

File: pkgs/dart_flutter_team_lints/lib/analysis_options.yaml [hunk 1]
```diff
@@ -17,7 +17,53 @@ include: package:lints/recommended.yaml
...
+    - await_only_futures
+    - hash_and_equals
+    - no_duplicate_case_values
+    - only_throw_errors
+    - throw_in_finally
+    - type_annotate_public_apis
+    - unawaited_futures
+    - unrelated_type_equality_checks
+
+    # optional?
```
Comment (natebosch) [inline pkgs/dart_flutter_team_lints/lib/analysis_options.yaml:56]: I think we should hold off on the optional group for now - these require a bit more investigation to be sure we can use them everywhere.

**Review by natebosch — APPROVED** [review]
consistency and correctness groups LGTM.

File: pkgs/dart_flutter_team_lints/lib/analysis_options.yaml (hunk 1 above)
Comment (devoncarew) [reply pkgs/dart_flutter_team_lints/lib/analysis_options.yaml]: Gotcha, will do.

File: pkgs/dart_flutter_team_lints/lib/analysis_options.yaml [hunk 2]
```diff
@@ -17,7 +17,38 @@ include: package:lints/recommended.yaml
```
Comment (kevmoo) [inline pkgs/dart_flutter_team_lints/lib/analysis_options.yaml:17]: Any desire to add the "language" section here and at least do `strict-casts`? @devoncarew ?

Comment (devoncarew) [reply pkgs/dart_flutter_team_lints/lib/analysis_options.yaml:17]: Here's the three strict options:

```
    strict-casts: true
    strict-inference: true
    strict-raw-types: true
```

cc @natebosch, but I'd wonder if these are things we generally want turned on, and, how they'd land w/ our current repos (would the settings cause work to update the repos).

Comment (kevmoo) [reply pkgs/dart_flutter_team_lints/lib/analysis_options.yaml:17]: strict-casts is almost always goodness, AFAIC – the others can be a bit noisy

Comment (natebosch) [reply pkgs/dart_flutter_team_lints/lib/analysis_options.yaml:17]: Yes, I don't see any reason we wouldn't want to turn on `strict-casts` everywhere.

**Pull request merged**